ding/utils/deprecation.py

```diff
@@ -0,0 +1,57 @@
+import functools
+import textwrap
+import warnings
+
+
+def deprecated(since, removed_in, up_to=None):
+    """Decorate a function to signify its deprecation.
+
+    Args:
+        since: The version when the function was first deprecated.
+        removed_in: The version when the function will be removed.
+        up_to: The new API users should use.
+    """
```

Comment:
maybe you can add an example in comments to better illustrate its effect. ([example](https://github.com/opendilab/DI-engine/blob/main/ding/model/common/encoder.py#L110))